### PR TITLE
Make controller tables display conditional on content

### LINKED ISSUE #857: Registered controller table appears even if empty
Screenshot 2021-02-03 at 14 21 14

### PR DESCRIPTION
**Pull request by barrymcgee**
## Done

Make controller tables display conditional on content

## QA

- Pull code
- Run `./run clean && ./run serve`
- Open http://0.0.0.0:8036/controllers
- Check tables display correctly

## Details

Fixes #857

### CODE REVIEW

**Comment by anthonydillon** [comment]
This does hide the table but you seem to not be aligned across the team and so you should gain approval from @clagom first.

Can we add tests for this functionality?

**Comment by clagom** [comment]
Thanks @anthonydillon 
@barrymcgee since apparently now we hide tables, this is inline with the rest of the pages. I found this though:
Screenshot 2021-02-04 at 13 11 40
Titles of empty tables should be hidden as well (not sure this can be part of this PR, I can open an issue)

**Comment by barrymcgee** [comment]
@clagom Good catch.. updated in this feature branch.

File: src/pages/ModelDetails/ModelDetails.js [hunk 1]
```diff
@@ -437,10 +437,17 @@ const ModelDetails = () => {
                       emptyStateMsg={"There are no relations in this model"}
                     />
                     {shouldShow("relations-title", activeView) && (
-                      <h5>
-                        Cross-model relations (
-                        {consumedTableRows.length + offersTableRows.length})
-                      </h5>
+                      <>
+                        {consumedTableRows.length > 0 ||
```
Comment (hatched) [inline src/pages/ModelDetails/ModelDetails.js:441]: There is quite a bit of logic here that's easy to get wrong in some updates, can you add a test for this plz

**Review by hatched — APPROVED** [review]
Approved but can you add that test plz 👍

File: src/pages/ModelDetails/ModelDetails.js (hunk 1 above)
Comment (barrymcgee) [reply src/pages/ModelDetails/ModelDetails.js:441]: The problem is the test data dump does not support CMR?

**Comment by barrymcgee** [comment]
Follow up issue to expand test data dump with these use cases so tests can be added at a later date: https://github.com/canonical-web-and-design/jaas-dashboard/issues/863

**Pull request merged**